src/basic/mount-util.c

```diff
@@ -495,6 +494,18 @@ int bind_remount_recursive(const char *prefix, bool ro, char **blacklist) {
         }
 }
 
+int bind_remount_recursive(const char *prefix, bool ro, char **blacklist) {
+        _cleanup_fclose_ FILE *proc_self_mountinfo = NULL;
+
+        proc_self_mountinfo = fopen("/proc/self/mountinfo", "re");
+        if (!proc_self_mountinfo) {
+                log_warning("Could not open /proc/self/mountinfo");
```

Comment:
in systemd out functions should either log all errors or none (that is, log at a log level != LOG_DEBUG. Logging at LOG_DEBUG is always OK). But with this patch we'd log some errors in this function, but others we wouldn't.

Given that bind_remount_recursive() never logged about anything and left that to the caller, bind_remount_recursive_with_mountinfo() should do the same. Hence, please either remove the log_warning() invocation, or downgrade it to LOG_DEBUG. If you opt for the latter, than you should log the actual error though, i.e. turn this into `return log_debug_errno(errno, "Could not open /proc/self/mountinfo: %m")`

Also see CODING_STYLE about this